### PR TITLE
[MAINT] Use rng fixture instead of check_random_state

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes https://github.com/nilearn/nilearn/issues/3926

### CODE REVIEW

File: nilearn/mass_univariate/tests/test_permuted_least_squares.py [hunk 1]
```diff
@@ -443,12 +436,11 @@ def test_permuted_ols_with_covar_with_intercept_in_confonding_vars(
 
 
 def test_permuted_ols_with_multiple_constants_and_covars(
-    design, random_state=RANDOM_STATE
+    design, rng, random_state=RANDOM_STATE
```
Comment (bthirion) [inline nilearn/mass_univariate/tests/test_permuted_least_squares.py:439]: Do we still need random_state here ?

File: nilearn/mass_univariate/tests/test_permuted_least_squares.py [hunk 2]
```diff
@@ -583,10 +573,8 @@ def test_permuted_ols_intercept_nocovar(random_state=RANDOM_STATE):
 
 
 def test_permuted_ols_intercept_statsmodels_withcovar(
-    random_state=RANDOM_STATE,
+    rng, random_state=RANDOM_STATE
```
Comment (bthirion) [inline nilearn/mass_univariate/tests/test_permuted_least_squares.py:576]: Same question here

File: nilearn/mass_univariate/tests/test_permuted_least_squares.py [hunk 3]
```diff
@@ -624,11 +612,9 @@ def test_permuted_ols_intercept_statsmodels_withcovar(
     assert output_intercept["t"].shape == (n_regressors, n_descriptors)
 
 
-def test_one_sided_versus_two_test(random_state=RANDOM_STATE):
+def test_one_sided_versus_two_test(rng, random_state=RANDOM_STATE):
```
Comment (bthirion) [inline nilearn/mass_univariate/tests/test_permuted_least_squares.py:615]: and here

File: nilearn/mass_univariate/tests/test_permuted_least_squares.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/mass_univariate/tests/test_permuted_least_squares.py]: OK I could remove this and also remove any use of RANDOM_STATE as a global in this test module.

But made realized the doc string of permuted_ols was not clear all types that random_state could take so I updated that.

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**